### PR TITLE
Port trace to rv32

### PR DESCRIPTION
**Pull request by luyahan**

### CODE REVIEW

File: src/execution/riscv32/simulator-riscv32.cc [hunk 1]
```diff
@@ -2590,13 +2585,6 @@ void Simulator::TraceMemRd(int32_t addr, T value, int32_t reg_value) {
                    reg_value, icount_, static_cast<int32_t>(value),
                    static_cast<uint32_t>(value), addr);
           break;
-        case 8:
```
Comment (qjivy) [inline src/execution/riscv32/simulator-riscv32.cc:2593]: If load double-precision float point （fld） is traced，the size would be 8

File: src/execution/riscv32/simulator-riscv32.cc [hunk 2]
```diff
@@ -2653,11 +2639,7 @@ void Simulator::TraceMemWr(int32_t addr, T value) {
         break;
       case 8:
         if (std::is_integral<T>::value) {
-          SNPrintF(trace_buf_,
-                   "                    (%" PRIu64 ")    int64:%" PRId64
-                   " uint64:%" PRIu64 " --> [addr: %" PRIx32 "]",
-                   icount_, static_cast<int64_t>(value),
-                   static_cast<uint64_t>(value), addr);
+          UNREACHABLE();
```
Comment (qjivy) [inline src/execution/riscv32/simulator-riscv32.cc:2642]: If store double-precision float point （fsd） is traced，the size would be 8

**Review by qjivy — CHANGES_REQUESTED** [review]

File: src/execution/riscv32/simulator-riscv32.cc (hunk 2 above)
Comment (luyahan) [reply src/execution/riscv32/simulator-riscv32.cc:2642]: `if (std::is_integral<T>::value) ` this is a int value

File: src/execution/riscv32/simulator-riscv32.cc (hunk 1 above)
Comment (luyahan) [reply src/execution/riscv32/simulator-riscv32.cc:2593]: `if (std::is_integral<T>::value) {` this is a int value

**Pull request merged**